### ISSUE TITLE
Add twisted.mail SMTP client example which demonstrates TLS and authentication

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2240]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2240|
|Type|enhancement|
|Created|2006-11-20 14:23:22Z|

Attachments:

* [smtpclient_tls.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/517/5173e941960d0a3fa26c06dc9d550adb5f16dcf8/9f6948845f408f3489f89b603677581147cad6f5.py) (4609 bytes) - added by exarkun on 2006-11-20 14:25:34Z - 
* [smtpclient_tls.2.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/517/5173e941960d0a3fa26c06dc9d550adb5f16dcf8/e1d7c8a6cb6a0c05704b0dbb206c0c070cc5c1de.py) (4153 bytes) - added by exarkun on 2006-11-27 21:54:16Z - New version of example - cleans up SSL handling a little
* [smtpclient_tls.3.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/517/5173e941960d0a3fa26c06dc9d550adb5f16dcf8/c4d7b9d7e36dd0630999533e5ec6ece5cd660bc3.py) (4619 bytes) - added by exarkun on 2006-11-28 23:44:07Z - Actually updated version

<details><summary>Searchable metadata</summary>

```
trac-id__2240 2240
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1164032602000000 1164032602000000
changetime__1165175045000000 1165175045000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Example attached.  Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @keturn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="keturn's avatar" src="https://avatars.githubusercontent.com/u/83819?s=50" width="50" height="50">](https://github.com/keturn)<a name="note_3"></a>|@keturn commented|
|-|-|

what's with verify=true being commented out on line 38?  Does that make the explanatory comment above it not so true?

**Comment by twisted-trac** [comment]
|<img alt="mailmesuf's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|mailmesuf commented|
|-|-|

Exarkun, thanks for this code it's a great example.

Could also you write a SMTP server which demonstates TLS and authentication?

I would be much appreciated.

grtz.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Replying to [acapnotic](#note_3):
> what's with verify=true being commented out on line 38?  Does that make the explanatory comment above it not so true?

Ummm, yea, forgot about that.  I'm just going to switch the example over to use the old SSL context stuff.  Setting up certificate verification is a separate thing and somewhat complex now, so it would just confuse this example.

**Comment by twisted-trac** [comment]
|<img alt="mailmesuf's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|mailmesuf commented|
|-|-|

To make the example work with Twisted version 2.4.0 i replaced the CertificateOptions class with ClientContextFactory.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_7"></a>|@radix set owner to @exarkun|
|-|-|

Ok

Need:
 80 columns

Want:
 Simpler English
 Unit Test / Rewrite as Doctest
 Pony

After 80col +1

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [18800]) Add SMTP TLS client example

Author: exarkun
Reviewer: radix
Fixes [#2240](https://github.com/twisted/twisted/issues/2240)

Add an example of using twisted.mail to send a message using authentication
and transport layer security.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
